pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -151,6 +138,10 @@ func (c *Reconciler) reconcileImageCache(ctx context.Context, rev *v1.Revision)
 
 func (c *Reconciler) reconcilePA(ctx context.Context, rev *v1.Revision) error {
 	ns := rev.Namespace
+
+	deploymentName := resourcenames.Deployment(rev)
+	deployment, _ := c.deploymentLister.Deployments(ns).Get(deploymentName)
```

Comment:
What would be the use case to ignore the error when retrieving the deployment?

Reply:
I think I did it for brevity to run the tests here - but we should handle it.

I believe the only error we could get there (since it's from the lister) is that it's not found - which isn't an issue since we handle nil deployments when creating the PA. I figured _we could_ miss having a deployment during first reconciliation if the informers watches are slow. Returning and error would just trigger the reconcilation to just retry.

We could strengthen the guarantee here and require the deployment be present. This would avoid a deployment going 'missing' and then the PA for a revision could then be marked reachable - which isn't logically correct.

I'll do an update.